docs/migrate_2.md

```diff
@@ -0,0 +1,111 @@
+# Migrating from rules_js 1.x to rules_js 2.0
+
+The 2.0 release is our first major release, where we've made breaking changes to keep the codebase healthy and adaptable to users needs. We expect to make major releases no more frequently than yearly.
+
+Users of rules_js 1.x should expect a fairly small effort required to migrate.
+
+You can follow the link to each rules_js pull request in the notes below.
+In many cases, updates to the testing folders (`e2e` or `examples`) are illustrative of the changes required.
+
+## Dependency versions
+
+Bazel 5 is no longer supported. Users must upgrade to Bazel 6 or greater. (#1589), (#1610)
+
+Minimum dependency versions have been increased:
+
+- `rules_nodejs`  6.1.0 (#1579)
+- `aspect_bazel_lib` 2.7.1 (#1580)
+
+## User facing changes
+
+Any rules_js user is likely to encounter these changes which require some edits to your code.
+
+### npm_translate_lock
+
+The label used to link a package, [`npm_package_target_name`](https://docs.aspect.build/rulesets/aspect_rules_js/docs/npm_translate_lock/#npm_package_target_name) has changed to `pkg` (#1684). You should change your `npm_package` rules to have `name = 'pkg'`, see [examples](https://github.com/aspect-build/rules_js/pull/1684/files#diff-d13f73189fcb63af69a6bfa0eb9cb61e71d6466d2eac4c1051674b73976a1cfe).
```

Comment:
"The default label" .. (you can change it back to `{dirname}` via `npm_package_target_name` if desired)